vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -262,87 +265,103 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
         } as IDotnetInstallationContext;
-        this.context.eventStream.post(new DotnetAcquisitionStarted(install, version, this.context.acquisitionContext?.requestingExtensionId));
+        context.eventStream.post(new DotnetAcquisitionStarted(install, version, context.acquisitionContext.requestingExtensionId));
         await acquisitionInvoker.installDotnet(installContext, install).catch((reason) =>
         {
             throw reason; // This will get handled and cast into an event based error by its caller.
         });
-        this.context.installationValidator.validateDotnetInstall(install, dotnetPath);
+        context.installationValidator.validateDotnetInstall(install, dotnetPath);
 
-        await this.removeMatchingLegacyInstall(installedVersions, version);
-        await this.tryCleanUpInstallGraveyard();
+        await this.removeMatchingLegacyInstall(context, installedVersions, version);
+        await this.tryCleanUpInstallGraveyard(context);
 
-        await this.installTracker.reclassifyInstallingVersionToInstalled(install);
+        await InstallTrackerSingleton.getInstance(context.eventStream, context.extensionState).reclassifyInstallingVersionToInstalled(context, install);
 
         return dotnetPath;
     }
 
-    private async checkForPartialInstalls(installKey : DotnetInstall)
+    private async checkForPartialInstalls(context: IAcquisitionWorkerContext, installKey : DotnetInstall)
     {
-        const installingVersions = await this.installTracker.getExistingInstalls(false);
+        const installingVersions = await InstallTrackerSingleton.getInstance(context.eventStream, context.extensionState).getExistingInstalls(false);
         const partialInstall = installingVersions.some(x => x.dotnetInstall.installKey === installKey.installKey);
 
         // Don't count it as partial if the promise is still being resolved.
         // The promises get wiped out upon reload, so we can check this.
-        if (partialInstall && this.installTracker.getPromise(installKey) === null)
+        if (partialInstall && InstallTrackerSingleton.getInstance(context.eventStream, context.extensionState).getPromise(installKey) === null)
         {
             // Partial install, we never updated our extension to no longer be 'installing'. Maybe someone killed the vscode process or we failed in an unexpected way.
-            this.context.eventStream.post(new DotnetAcquisitionPartialInstallation(installKey));
+            context.eventStream.post(new DotnetAcquisitionPartialInstallation(installKey));
 
             // Delete the existing local files so we can re-install. For global installs, let the installer handle it.
-            await this.uninstallLocalRuntimeOrSDK(installKey);
+            await this.uninstallLocalRuntimeOrSDK(context, installKey);
+        }
+    }
+
+    public async tryCleanUpInstallGraveyard(context: IAcquisitionWorkerContext) : Promise<void>
```

Comment:
Does this graveyard contain... g.. g... g.... ghosts?!?! 👻😱

Reply:
Ghosts of .NET Installs, yes 😄 Maybe the WIX Ghost, too